bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/jackson/DefaultMethodSkippingModuleProvider.java

```diff
@@ -49,8 +51,15 @@ public DefaultMethodSkippingModuleProvider() {
                     if (member instanceof Method) {
                         final Method method = (Method) member;
                         if (method.isDefault()) {
-                            // only exclude default methods if they are defined on interfaces from the core components
-                            return !method.getDeclaringClass().getName().startsWith(PACKAGE_CORE_COMPONENTS);
+                            try {
+                                // only exclude default methods if they are defined on interfaces from the core components
+                                String className = beanDesc.getBeanClass().getMethod(method.getName()).getDeclaringClass().getName();
+                                return !className.startsWith(PACKAGE_CORE_COMPONENTS) ||
+                                    (className.startsWith(PACKAGE_CORE_COMPONENTS) &&
```

Comment:
This is redundant:
- if `true`: `!true || (true && X) == X`
- if `false`: `!false || (false && X) == true`
Could be simplified to `!className.startsWith(PACKAGE_CORE_COMPONENTS) || className.contains(PACKAGE_IMPL_INTERNAL)`

Reply:
updated in 5e845b93dd42bdb43be28085f632a5a58c33b334